clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -366,8 +383,16 @@ public void close() {
         }
     }
 
+    // visible for testing
+    void invokeCompletedOffsetCommitCallbacks() {
+        for (OffsetCommitCompletion completion : completedOffsetCommits)
+            completion.invoke();
+        completedOffsetCommits.clear();
+    }
+
+    public synchronized void commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
```

Comment:
What exactly is this synchronized for? I think this can only be called by the user thread, so is this just for synchronizing `completedOffsetCommits`?